File: Sorts/BubbleSort/BubbleSort.c

//=====INCLUDES=====//
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include "../Utils/Generator.h"

//=====DEFINE=====//
#define LEN 100

//=====PROTOTYPES=====//
void bubbleSorti(int arr[], int first, int last);
void swapi(int arr[], int a, int b);

void bubbleSortd(double arr[], int first, int last);
void swapd(double arr[], int a, int b);

//=====MAIN=====//
int main(char argc, char *args) {
    double arr[LEN];
    for(int i = 0; i < 100; i++) {
        generated(arr, LEN, 10000);
        bubbleSortd(arr, 0, LEN);
        printf("%i\n", verifyd(arr, LEN));
    }

    return 0;
}

//=====METODOS=====//
void bubbleSorti(int arr[], int first, int last) {
    for(int i = first; i < last - 1; i++) for(int j = first; j < last - i - 1; j++) if(arr[j] > arr[j + 1]) swapi(arr, j, j + 1);
}

void swapi(int arr[], int a, int b) {
    int aux = arr[a];
    arr[a] = arr[b];
    arr[b] = aux;
}

void bubbleSortd(double arr[], int first, int last) {
    for(int i = first; i < last - 1; i++) for(int j = first; j < last - i - 1; j++) if(arr[j] > arr[j + 1]) swapd(arr, j, j + 1);
}

void swapd(double arr[], int a, int b) {
    double aux = arr[a];
    arr[a] = arr[b];
    arr[b] = aux;
} 

File: Sorts/BubbleSort/BubbleSort.java
package Sorts.BubbleSort;

import Sorts.Utils.Generator;

/**
 * @author Fernando Campos Silva Dal Maria
 * @version 1.0
 */
public class BubbleSort {

    public static void main(String[] args) {
        double[] arr = new double[100];
        
        for(int i = 0; i < 100; i++) {
            Generator.generate(arr, 10000);
            bubbleSort(arr, 0, arr.length);
            System.out.println(Generator.verify(arr)); 
        }
    }

    public static void bubbleSort(int[] arr, int first, int last) {
        for(int i = first; i < last - 1; i++) for(int j = first; j < last - i - 1; j++) if(arr[j] > arr[j + 1]) swap(arr, j, j + 1);
    }

    public static void swap(int[] arr, int a, int b) {
        int aux = arr[a];
        arr[a] = arr[b];
        arr[b] = aux;
    }

    public static void bubbleSort(double[] arr, int first, int last) {
        for(int i = first; i < last - 1; i++) for(int j = 0; j < last - i - 1; j++) if(arr[j] > arr[j + 1]) swap(arr, j, j + 1);
    }

    public static void swap(double[] arr, int a, int b) {
        double aux = arr[a];
        arr[a] = arr[b];
        arr[b] = aux;
    }

}